tool/tsh/common/proxy.go

```diff
@@ -317,16 +320,41 @@ func onProxyCommandApp(cf *CLIConf) error {
 		return trace.Wrap(err)
 	}
 
-	appCerts, err := loadAppCertificateWithAppLogin(cf, tc, cf.AppName)
+	app, err := getRegisteredApp(cf, tc)
 	if err != nil {
 		return trace.Wrap(err)
 	}
 
-	app, err := getRegisteredApp(cf, tc)
+	profile, err := tc.ProfileStatus()
+	if err != nil {
+		return trace.Wrap(err)
+	}
+
+	routeToApp, err := getRouteToApp(cf, tc, profile, app)
 	if err != nil {
 		return trace.Wrap(err)
 	}
 
+	opts := []alpnproxy.LocalProxyConfigOpt{
+		alpnproxy.WithALPNProtocol(alpnProtocolForApp(app)),
+		alpnproxy.WithClusterCAsIfConnUpgrade(cf.Context, tc.RootClusterCACertPool),
+		alpnproxy.WithMiddleware(client.NewAppCertChecker(tc, routeToApp, nil)),
+	}
+
+	// If MFA is not required, try to load existing certs. If MFA is required,
```

Comment:
I'm not an expert on this part of the codebase, but I have mixed feelings about keeping this logic in.

When would that be the case even? I guess someone would need to do `tsh app login` followed by `tsh proxy app`?

What I'm trying to say is that if we don't have tests for this scenario, keeping the logic for it around seems like a maintenance burden and possible source of bugs.

Reply:
This was meant to maintain feature parity with `tsh proxy kube|db` but you're right that it's not necessary. It actually makes more sense to me to have `tsh proxy app` create a new session separate from any ongoing `tsh app login` credentials, so the proxy session can be its own. Thanks for raising this.